src/4D_api.cpp

```diff
@@ -966,6 +966,423 @@ std::string pj_add_type_crs_if_needed(const std::string& str)
+            && data[iii] != HUGE_VAL
+        )
+            negative_max = data[iii];
+        // track general max value
+        if ((data[iii] > max_value || max_value == HUGE_VAL) && data[iii] != HUGE_VAL)
+            max_value = data[iii];
+    }
+    if (crossed_meridian_count == 2)
+        return negative_max;
+    else if (crossed_meridian_count == 4)
+        // bounds extends beyond -180/180
+        return 180;
+    return max_value;
+}
+
+
+// ---------------------------------------------------------------------------
+// Check if the original projected bounds contains
+// the north pole.
+// This assumes that the destination CRS is geographic.
+static bool contains_north_pole(
+    PJ* projobj,
+    PJ_DIRECTION pj_direction,
+    const double left,
+    const double bottom,
+    const double right,
+    const double top
+) {
+    double pole_y = 90;
+    double pole_x = 0;
+    proj_trans_generic(
+        projobj,
+        opposite_direction(pj_direction),
+        &pole_x, sizeof(double), 1,
+        &pole_y, sizeof(double), 1,
+        nullptr, sizeof(double), 0,
+        nullptr, sizeof(double), 0
+    );
+    if (left < pole_x && pole_x < right && top > pole_y && pole_y > bottom)
+        return true;
+    return false;
+}
+
+
+// ---------------------------------------------------------------------------
+// Check if the original projected bounds contains
+// the south pole.
+// This assumes that the destination CRS is geographic.
+static bool contains_south_pole(
+    PJ* projobj,
+    PJ_DIRECTION pj_direction,
+    const double left,
+    const double bottom,
+    const double right,
+    const double top
+) {
+    double pole_y = -90;
+    double pole_x = 0;
+    proj_trans_generic(
+        projobj,
+        opposite_direction(pj_direction),
+        &pole_x, sizeof(double), 1,
+        &pole_y, sizeof(double), 1,
+        nullptr, sizeof(double), 0,
+        nullptr, sizeof(double), 0
+    );
+    if (left < pole_x && pole_x < right && top > pole_y && pole_y > bottom)
+        return true;
+    return false;
+}
+
+
+// ---------------------------------------------------------------------------
+
+/** \brief Transform boundary,
+ *
+ * Transform boundary densifying the edges to account for nonlinear
+ * transformations along these edges and extracting the outermost bounds.
+ *
+ * If the destination CRS is geographic and right < left then the bounds
+ * crossed the antimeridian. In this scenario there are two polygons,
+ * one on each side of the antimeridian. The first polygon should be
+ * constructed with (left, bottom, 180, top) and the second with
+ * (-180, bottom, top, right).
+ *
+ * When projecting from polar projections to geographic,
+ * lon, lat output order is required.
+ *
+ * @param P The PJ object representing the transformation.
+ * @param direction The direction of the transformation.
+ * @param left The input left bounding coordinate.
```

Comment:
we should probably mention that left is according to the first axis of the source CRS of P* (or the target CRS if using the reverse direction). Same  kindremark for the other 7 arguments

Reply:
and the left / bottom / right / top names aren't probably appropriate if we just follow the axis order. xmin, ymin, xmax, ymax ? where xmin is a latitude in EPSG:4326 ?